### ISSUE TITLE
Decimal literal suffix of 0 (`2.` vs. `2.0`)

### ISSUE THREAD

**Issue by BD103** [rust-lang/rust-clippy#13022]
### What it does

This lint checks if a floating-point number with no decimals explicitly writes `.0` or just `.`

In the Bevy game engine there are a lot of examples that use floating-point numbers. Because so many people contribute to these examples, there's a large mix between literals like `1.0` and `1.`. [Here's the most egregious example.](https://github.com/bevyengine/bevy/blob/8a7d3ce461058bad2e0aaa608353abe2cd4da6d0/examples/ui/ui.rs#L235) Having a lint that picks one form and denies the other would be great for consistency, though it should probably be within the pedantic or nursery groups.

I'm not sure how a user would choose which form they prefer, though, since exclusive lints are probably not a good idea.

### Advantage

- Increases consistency of writing floating point numbers by only allowing one form.

### Drawbacks

- It might be difficult to implement, because there's no necessarily "correct" form. Both have their own benefits, meaning you would have to let the user decide.

### Example

```rust
let x = 1.0;
```

Could be written as:

```rust
let x = 1.;
```

And vice-versa.

**Comment by Alexendoo** [comment]
Looks like there's a pending effort to get this into rustfmt - https://github.com/rust-lang/rustfmt/pull/5834

**Comment by BD103** [comment]
Awesome, I didn't see that in my initial search. Thanks for pointing it out!
